Jenkinsfile

```diff
@@ -1,90 +1,115 @@
+            ]) {
             // Obtain values for the server url, project name and folder name from a Jenkins config file, which have the ID
             // that matches the name of the current pipeline job.
-            configFileProvider([configFile(fileId: env.JOB_NAME, targetLocation: 'test.props')]) {
-                def props = readProperties file: 'test.props'
-                echo "Running tests for server ${props.server}, project ${props.project}, folder ${props.folder}"
-                def credentials = [Email:user,Password:pwd,ExecutionOnly:true]
-                String cookie = getAuthenticationCookie(props.server, credentials)
-                def tests = getTests(props.server, props.project, props.folder, cookie)
-                def testJobs = [:]
-                for (testname in tests) {
-                    def tn = testname
-                    echo "Scheduling ${tn}"
-                    testJobs[tn] = {
+            String TEST_SERVICE_URL = "https://tests.api.usemango.co.uk"
+            String SCRIPTS_SERVICE_URL = "https://scripts.api.usemango.co.uk"
+            String APP_WEBSITE_URL = "https://app.usemango.co.uk"     
+            echo "Running tests in project ${params['Project']} with tags ${params['Tags']}"
+            def tests = getTests(TEST_SERVICE_URL)
+            def testJobs = [:]
+            def testResults = [:]
+            tests.eachWithIndex { id, index ->
+                echo "Scheduling ${id}"
+                    testJobs[id] = {
                         node('usemango') {
-                            try {
-                                unstash 'scripts'
-                                bat "\"%programfiles(x86)%\\Infuse Consulting\\useMango\\App\\MangoMotor.exe\" -s ${props.server} -p ${props.project} --testname \"${tn}\" -e ${user} -a ${pwd}"
-                            }
-                            finally {                                
-                                junit \"%PROGRAMDATA%\\useMango\\logs\\junit.xml\"
+                            wrap([$class: "MaskPasswordsBuildWrapper", varPasswordPairs: [[password: "$Apikey"]]]) {
+                                dir ("${env.WORKSPACE}\\${tests[index]}") {
+                                    deleteDir()
+                                }
+                                dir("${env.WORKSPACE}\\${tests[index]}") {
+                                    bat "curl -s --create-dirs -L -D \"${env.WORKSPACE}\\${tests[index]}\\response.txt\" -X GET \"${SCRIPTS_SERVICE_URL}/tests/${tests[index]}\" -H \"Authorization: APIKEY $Apikey\" --output \"${env.WORKSPACE}\\${tests[index]}\\${tests[index]}.pyz\""
+                                    String httpCode = powershell(returnStdout: true, script: "Write-Output (Get-Content \"${env.WORKSPACE}\\${tests[index]}\\response.txt\" | select -First 1 | Select-String -Pattern '.*HTTP/1.1 ([^\\\"]*) *').Matches.Groups[1].Value")                             
```

Comment:
Do we need to specify the full path? We should be able to use relative paths since we're using `dir` function